spyder/plugins/ipythonconsole.py

```diff
@@ -619,8 +619,9 @@ def __init__(self, parent, testing=False):
 
         # Create temp dir on testing to save kernel errors
         if self.testing:
-            if not osp.isdir(programs.TEMPDIR):
-                os.mkdir(programs.TEMPDIR)
+            if not osp.isdir(osp.join(programs.TEMPDIR, u'測試', u'اختبار')):
+                os.makedirs(osp.join(programs.TEMPDIR, u'測試', u'اختبار'))

```

Comment:
@dalthviz, I don't like this test directory to be part of our sourcecode.

Please add a new parameter to `__init__` called `test_dir=TEMPDIR`. With that this should look like:

```python
if self.testing:
    if not osp.isdir(test_dir):
        os.makedirs(test_dir)
```

Then in our tests you can change that parameter by this directory containing Chinese and Arabic characters.